js/adapt-contrib-bookmarking.js

```diff
@@ -177,52 +175,39 @@ class Bookmarking extends Backbone.Controller {
       return;
     }
 
-    this.watchViewIds = pageView.model.findDescendantModels(this.bookmarkLevel + 's').map(desc => desc.get('_id'));
-
-    this.listenTo(Adapt, this.bookmarkLevel + 'View:postRender', this.captureViews);
     this.listenToOnce(Adapt, 'remove', this.releaseViews);
 
     $(window).on('scroll', this._onScroll);
   }
 
-  captureViews(view) {
-    this.watchViews.push(view);
-  }
-
   setLocationID(id) {
     if (this.currentLocationID === id) return;
     Adapt.offlineStorage.set('location', id);
     this.currentLocationID = id;
   }
 
   releaseViews() {
-    this.watchViews.length = 0;
-    this.watchViewIds.length = 0;
     this.stopListening(Adapt, 'remove', this.releaseViews);
-    this.stopListening(Adapt, this.bookmarkLevel + 'View:postRender', this.captureViews);
     $(window).off('scroll', this._onScroll);
   }
 
   checkLocation() {
+    if (!this.currentLocationID) return;
```

Comment:
one last thing: just wondering why we need this check..?

Reply:
we don't really